packages/cln_plugin/lib/src/json_rpc/request.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'request.g.dart';
+
+@JsonSerializable()
+class Request {
+  String jsonrpc;
+  // The spec allows to set the id as a String, but core lightning uses int,
+  // so we strictly use int to make the parsing less painful.
+  int id;
+  String method;
+
+  /// TODO add docs comments
+  Map<String, dynamic> params;
+
+  Request(
+      {required this.jsonrpc,
```

Comment:
Maybe we can enforce a check right here if the string is exactly `2.0` or not.

Reply:
the check need to done by the server usually and not the client, the check can be done but it is not out job :)

Reply:
Cool! :)